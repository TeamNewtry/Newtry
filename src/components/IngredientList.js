import {Text, View, StyleSheet} from 'react-native';
import Unorderedlist from 'react-native-unordered-list';
import React, {useContext} from 'react';
import {LocalizationContext} from './Translations';

const splitIngredients = ingredients => {
  ingredients = ingredients ?? '';
  const regex = /[^,([]+([([].*?[)\]]+)?/g;
  const results = [...ingredients.matchAll(regex)];
  return results
    .map(result => result[0].trim())
    .filter(result => result.length > 1);
};
const IngredientList = ing => {
  const {translations} = useContext(LocalizationContext);
  return (
    <View>
      <Text style={styles.title}>
        {translations['nutrition.ingredients.header']}
      </Text>
      {splitIngredients(ing.data).map(element => {
        return (
          <Unorderedlist key={element}>
            <Text style={styles.description}>{element}</Text>
          </Unorderedlist>
        );
      })}
    </View>
  );
};

export default IngredientList;

const styles = StyleSheet.create({
  title: {
    fontSize: 20,
    paddingTop: 10,
    fontFamily: 'Comfortaa',
  },
  description: {
    fontSize: 14,
    paddingBottom: 1,
    fontFamily: 'Comfortaa',
  },
});
