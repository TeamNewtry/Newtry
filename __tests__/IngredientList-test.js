import 'react-native';
import React from 'react';
import data from '../__mocks__/productMock';
import IngredientList from '../src/components/IngredientList';
import renderer from 'react-test-renderer';

describe('testing ingredients list', () => {
  it('renders empty view when no data is passed', () => {
    const tree = renderer.create(<IngredientList />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders list for passed data', () => {
    const tree = renderer
      .create(<IngredientList data={data.ingredients} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
