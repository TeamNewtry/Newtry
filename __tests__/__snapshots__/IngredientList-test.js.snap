// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing ingredients list renders empty view when no data is passed 1`] = `
<View>
  <Text
    style={
      Object {
        "fontFamily": "Comfortaa",
        "fontSize": 20,
        "paddingTop": 10,
      }
    }
  >
     Ingredients
  </Text>
</View>
`;

exports[`testing ingredients list renders list for passed data 1`] = `
<View>
  <Text
    style={
      Object {
        "fontFamily": "Comfortaa",
        "fontSize": 20,
        "paddingTop": 10,
      }
    }
  >
     Ingredients
  </Text>
</View>
`;
